### PR TITLE
Make signal handling resilient to asynchronous calls

### PR DESCRIPTION
**Pull request by Kr1sso**
Signal handling happens asynchronously on Unix machines, hence handling code has to be able to handle reentrant scenarios. Dispatching signal handlers to a global queue via GCD on macOS helps here and makes the signal handling thread safe.

### CODE REVIEW

File: Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/CoreDumpTester.xcscheme [hunk 1]
```diff
@@ -82,10 +82,10 @@
       </BuildableProductRunnable>
    </ProfileAction>
    <AnalyzeAction
-      buildConfiguration = "Debug">
+      buildConfiguration = "release">
    </AnalyzeAction>
    <ArchiveAction
-      buildConfiguration = "Release"
+      buildConfiguration = "release"
```
Comment (almiliMSFT) [inline Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/CoreDumpTester.xcscheme]: just curious: are any of these lower vs upper casing changes significant?

File: Public/Src/Sandbox/MacOs/Interop/Posix/process.c [hunk 2]
```diff
@@ -226,3 +226,48 @@ bool SetupProcessDumps(const char *logsDirectory, /*out*/ char *buffer, size_t b
...
+    dispatch_source_set_event_handler(source, ^{
+        sigCrashHandler(SIGABRT);
+    });
+    dispatch_resume(source);
+
+    source = dispatch_source_create(DISPATCH_SOURCE_TYPE_SIGNAL, SIGSEGV, 0, dispatch_get_global_queue(0, 0));
+    dispatch_source_set_event_handler(source, ^{
+        sigCrashHandler(SIGSEGV);
+    });
+    dispatch_resume(source);
```
Comment (almiliMSFT) [inline Public/Src/Sandbox/MacOs/Interop/Posix/process.c]: DRY:
```c
    int signals[] = { SIGBUS, SIGILL, SIGHUP, SIGABRT, SIGSEGV };
    int signalsLength = sizeof(signals)/sizeof(signals[0]);
    for (int i = 0; i < signalsLength; i++)
    {
        int sig = signals[i];
        sigaction(sig, &action, NULL);
        dispatch_source_t source = dispatch_source_create(DISPATCH_SOURCE_TYPE_SIGNAL, sig, 0, dispatch_get_global_queue(0, 0));
        dispatch_source_set_event_handler(source, ^{
            sigCrashHandler(sig);
        });
        dispatch_resume(source);
    }
```

File: Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/InteropLibrary.xcscheme [hunk 3]
```diff
@@ -71,10 +71,10 @@
       </MacroExpansion>
    </ProfileAction>
    <AnalyzeAction
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/InteropLibrary.xcscheme]: >AnalyzeAction [](start = 4, length = 13)

Remove element? It seems empty...

File: Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/InteropLibrary.xcscheme [hunk 4]
```diff
@@ -23,7 +23,7 @@
       </BuildActionEntries>
    </BuildAction>
    <TestAction
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/InteropLibrary.xcscheme]: >TestAction [](start = 4, length = 10)

Intentional swap between debug and release?

File: Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/CoreDumpTester.xcscheme [hunk 5]
```diff
@@ -82,10 +82,10 @@
       </BuildableProductRunnable>
    </ProfileAction>
    <AnalyzeAction
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/CoreDumpTester.xcscheme]: Remove, not used...

File: Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/AriaLibrary.xcscheme [hunk 6]
```diff
@@ -71,10 +71,10 @@
       </MacroExpansion>
    </ProfileAction>
    <AnalyzeAction
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/AriaLibrary.xcscheme]: >AnalyzeAction [](start = 4, length = 13)

Remove, empty and flipped between debug & release

File: Public/Src/Sandbox/MacOs/Sandbox/Sandbox.xcodeproj/xcshareddata/xcschemes/BuildXLSandbox.xcscheme [hunk 7]
```diff
@@ -71,10 +71,10 @@
       </MacroExpansion>
    </ProfileAction>
    <AnalyzeAction
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/Sandbox/Sandbox.xcodeproj/xcshareddata/xcschemes/BuildXLSandbox.xcscheme]: empty

File: Public/Src/Sandbox/MacOs/Sandbox/Sandbox.xcodeproj/xcshareddata/xcschemes/SandboxMonitor.xcscheme [hunk 8]
```diff
@@ -17,13 +17,13 @@
                BlueprintIdentifier = "F51A2BF82190C67500880752"
                BuildableName = "SandboxMonitor"
                BlueprintName = "SandboxMonitor"
-               ReferencedContainer = "container:BuildXLSandbox.xcodeproj">
+               ReferencedContainer = "container:Sandbox.xcodeproj">
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/Sandbox/Sandbox.xcodeproj/xcshareddata/xcschemes/SandboxMonitor.xcscheme]: >Sandbox [](start = 48, length = 7)

intentional? Didn't see the xcode project renamed...

File: Public/Src/Sandbox/MacOs/Sandbox/Sandbox.xcodeproj/xcshareddata/xcschemes/SandboxMonitor.xcscheme [hunk 9]
```diff
@@ -87,27 +87,17 @@
...
-         <BuildableReference
-            BuildableIdentifier = "primary"
-            BlueprintIdentifier = "F51A2BF82190C67500880752"
-            BuildableName = "SandboxMonitor"
-            BlueprintName = "SandboxMonitor"
-            ReferencedContainer = "container:BuildXLSandbox.xcodeproj">
-         </BuildableReference>
-      </BuildableProductRunnable>
    </ProfileAction>
    <AnalyzeAction
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/Sandbox/Sandbox.xcodeproj/xcshareddata/xcschemes/SandboxMonitor.xcscheme]: empty and flipped

File: Public/Src/Sandbox/MacOs/Sandbox/Src/Resources/Info.plist [hunk 10]
```diff
@@ -7,7 +7,7 @@
 	<key>CFBundleExecutable</key>
 	<string>$(EXECUTABLE_NAME)</string>
 	<key>CFBundleIdentifier</key>
-	<string>$(BUILDXL_BUNDLE_IDENTIFIER)</string>
+	<string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
```
Comment (dannyvv) [inline Public/Src/Sandbox/MacOs/Sandbox/Src/Resources/Info.plist]: >PRODUCT_BUNDLE_IDENTIFIER [](start = 11, length = 25)

didn't see corresponding define change...

File: Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/CoreDumpTester.xcscheme (hunk 1 above)
Comment (Kr1sso) [reply Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/CoreDumpTester.xcscheme]: I guess when you were introducing the xcconfig files for the debug / release configuration you changed configuration names maybe? I didn't look at the history - but Xcode takes the name of the configuration which is lowercase in both cases and had a reference to uppercase, resulting in three entries to chose from although only two were present. Also the release xcconfig was the default for debug builds, which made the builds "release" flavored and bxl complain about flavor mismatch.

File: Public/Src/Sandbox/MacOs/Sandbox/Sandbox.xcodeproj/xcshareddata/xcschemes/SandboxMonitor.xcscheme (hunk 8 above)
Comment (Kr1sso) [reply Public/Src/Sandbox/MacOs/Sandbox/Sandbox.xcodeproj/xcshareddata/xcschemes/SandboxMonitor.xcscheme]: The scheme was referencing both, the removed non-existent one and the right one (Sandbox.xcodeproj). The latter was already present and the former just a dead reference.

File: Public/Src/Sandbox/MacOs/Sandbox/Src/Resources/Info.plist (hunk 10 above)
Comment (Kr1sso) [reply Public/Src/Sandbox/MacOs/Sandbox/Src/Resources/Info.plist]: `PRODUCT_BUNDLE_IDENTIFIER` is the default entry for plist files, we assign the our custom identifier value to it inside the Sandbox project:

```PRODUCT_BUNDLE_IDENTIFIER = "$(BUILDXL_BUNDLE_IDENTIFIER)";```

File: Public/Src/Sandbox/MacOs/Interop/Posix/process.c (hunk 2 above)
Comment (Kr1sso) [reply Public/Src/Sandbox/MacOs/Interop/Posix/process.c]: We all know I like repeating myself ;) 👍

File: Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/InteropLibrary.xcscheme (hunk 3 above)
Comment (Kr1sso) [reply Public/Src/Sandbox/MacOs/Interop/Interop.xcodeproj/xcshareddata/xcschemes/InteropLibrary.xcscheme]: All of this is auto generated by the Xcode IDE depending on settings / schema changes. I'd rather not manually tweak that data. We had a small misconfiguration that got fixed, but thanks for the sanity checks 👍

**Pull request merged**